Source/Scene/BillboardCollection.js

```diff
@@ -1015,6 +1053,41 @@ define([
      * @exception {RuntimeError} image with id must be in the atlas.
      */
     BillboardCollection.prototype.update = function(context, frameState, commandList) {
+        var i;
+        var j;
+
+        var startTime = getTimestamp();
+        var timeSlice = this._clampTimeSlice;
+        var endTime = startTime + timeSlice;
+
+        var tileList = this._renderedTileList;
+        while (tileList.length > 0) {
+            var tile = tileList[0];
```

Comment:
Why not use `Queue`?  For bonus, pull in (and finish) some of the simple optimizations I [started adding to it](https://github.com/AnalyticalGraphicsInc/cesium/compare/gfx#diff-c836627f72a2d12d57127f571f0337acR2).